### PR TITLE
Aave V1 filter

### PR DESCRIPTION
**Pull request by elenadimitrova**
Allow calls to deposit ETH via `LendingPool.deposit(address,uint256,uint16)`

And redeeming aETH via `aToken.redeem(uint256)`

Integration tests working with mainnet contracts:
```
LendingPoolCore: 0x3dfd23A6c5E8BbcFc9581d2E864a68feb6a076d3
LendingPool     0x398eC7346DcD622eDc5ae82352F02bE94C62d119
aETH Token      0x3a3A65aAb0dd2A17E3F1947bA16138cd37d08c04
```

Deployment configuration setup with Ropsten instances:
```
LendingPoolCore: 0x4295ee704716950a4de7438086d6f0fbc0ba9472
LendingPool     0x9E5C7835E4b13368fd628196C4f1c6cEc89673Fa
aETH Token      0x2433A1b6FcF156956599280C3Eb1863247CFE675
```

### CODE REVIEW

File: utils/config-schema.json [hunk 1]
```diff
@@ -122,6 +122,12 @@
           "properties": {
             "contract": {
               "$ref": "#/definitions/ethaddress"
+            },
+            "lendingPool": {
+              "$ref": "#/definitions/ethaddress"
+            },
+            "aaveToken": {
```
Comment (olivdb) [inline utils/config-schema.json:129]: There are several aTokens so I think this should be an array `aTokens` of addresses.

File: deployment/7_deploy_2.5.js [hunk 2]
```diff
@@ -203,6 +205,16 @@ const main = async () => {
...
 
+  // Aave V1
+  console.log("Deploying AaveV1");
+  const AaveV1FilterWrapper = await AaveV1Filter.new();
+  console.log(`Deployed AaveV1Filter at ${AaveV1FilterWrapper.address}`);
+  await DappRegistryWrapper.addDapp(0, config.defi.aave.lendingPool, AaveV1FilterWrapper.address);
+
+  const AaveETHTokenFilterWrapper = await AaveETHTokenFilter.new();
+  console.log(`Deployed AaveETHTokenFilter at ${AaveETHTokenFilterWrapper.address}`);
+  await DappRegistryWrapper.addDapp(0, config.defi.aave.aaveToken, AaveETHTokenFilterWrapper.address);
```
Comment (olivdb) [inline deployment/7_deploy_2.5.js:216]: I think we should add that filter for each aToken in `config.defi.aave.aTokens`

File: contracts/infrastructure/dapp/AaveETHTokenFilter.sol [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.12;
+
+import "./BaseFilter.sol";
+
+contract AaveETHTokenFilter is BaseFilter {
```
Comment (olivdb) [inline contracts/infrastructure/dapp/AaveETHTokenFilter.sol:21]: Why `AaveETHTokenFilter`? Should we name this filter `AaveV1ATokenFilter` (and the other one `AaveV1LendingPoolFilter` maybe)?

File: deployment/7_deploy_2.5.js [hunk 4]
```diff
@@ -203,6 +205,16 @@ const main = async () => {
   console.log(`Deployed UniswapV2Filter at ${UniswapV2FilterWrapper.address}`);
   await DappRegistryWrapper.addDapp(0, config.defi.uniswap.unizap, UniswapV2FilterWrapper.address);
 
+  // Aave V1
+  console.log("Deploying AaveV1");
```
Comment (olivdb) [inline deployment/7_deploy_2.5.js:209]: I think we should also authorise the [LendingPoolCore] (https://etherscan.io/address/0x3dfd23A6c5E8BbcFc9581d2E864a68feb6a076d3) contract (used as the target of an ERC20 approval prior to depositing the ERC20 into the lending pool) with our OnlyApproveFilter.

Comment (olivdb) [reply deployment/7_deploy_2.5.js:209]: And we could also add that "ERC20 deposit" use case to our tests as well

**Review by olivdb — COMMENTED** [review]
Could we add simple unit tests of this filter (and the Lido filter)? I think it's fine to just use dumb mock contracts that don't do anything, just so we don't reduce our `infrastructure` coverage score.

File: utils/config-schema.json (hunk 1 above)
Comment (elenadimitrova) [reply utils/config-schema.json:129]: Okay, only did that because Tom was saying we only support aETH atm. Is that not the case?

**Comment by elenadimitrova** [comment]
I was thinking of doing the opposite for testing, i.e. move all filters tests to be integration and run coverage on those

**Comment by olivdb** [comment]
> I was thinking of doing the opposite for testing, i.e. move all filters tests to be integration and run coverage on those. It'll shrink out lib and test contracts estate significantly and also reduce risk we get out of date with production and also flag integration issues early.

I don't think we should get rid of (non-integration) unit tests for filters. Unit tests run faster and are not at risk of breaking when some state changes in a third party contract. Also, in some cases we are going to have a proactive integration of a protocol that is not yet deployed on any testnet or mainnet (as is the case for Paraswap).

**Comment by elenadimitrova** [comment]
In terms of speed, integration tests are faster end-to-end because a) they don't need us to compile mocked contracts and b) we reuse existing instances rather than create new ones. Proactive integration is fine, but that's more the exception than the rule for filters development.

**Review by olivdb — APPROVED** [review]

**Pull request merged**